### PR TITLE
add search module in AD

### PR DESCRIPTION
**Pull request by pbivic**

### CODE REVIEW

File: MicrosoftActiveDirectory/action_search.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        "description": "The starting point an LDAP server uses when searching for users authentication within your Directory. (e.g DC=example-domain,DC=com)",
+        "type": "string"
+      },
+      "attributes": {
+        "title": "Attributes",
+        "description": "Attributes you want to retrieve (default will be ALL) ",
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
```
Comment (squioc) [inline MicrosoftActiveDirectory/action_search.json:26]: In order to express that the property can be nullable:
```suggestion
        "anyOf": [
          {
            "type": "array",
            "items": {
              "type": "string"
            }
          },
          {"type": "null"}
        ],
        "default": null
```

**Review by squioc — CHANGES_REQUESTED** [review]
LGTM. A little change to apply in the definition of the action.

File: MicrosoftActiveDirectory/tests/test_search.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    return a
+
+
+@pytest.fixture
+def one_user_dn():
+    return [["CN=integration_test,CN=Users,DC=lab,DC=test,DC=com", 512]]
+
+def test_search():
```
Comment (TOUFIKIzakarya) [inline MicrosoftActiveDirectory/tests/test_search.py:24]: I'm not sure that this test will pass. I think you need to mock the search method.

**Review by TOUFIKIzakarya — COMMENTED** [review]
Good job paul. Thanks for that.
just a little remark about tests

**Review by TOUFIKIzakarya — APPROVED** [review]
LGFM

**Review by squioc — APPROVED** [review]
LGTM

**Pull request merged**